lib/google_crawler_web/templates/keyword/index.html.eex

```diff
@@ -1,34 +1,59 @@
-<h1>Listing Keywords</h1>
+  <div class="row">
+    <div class="filter-form col-3">
+      <%= form_for @search_form, Routes.keyword_path(@conn, :index),[method: :get, as: "filter"], fn f -> %>
+        <div class="form-group">
+          <%= label :term, "Search Term" %>
+          <%= search_input f, :term, [id: "filter_term"] %>
+        </div>
 
-<%= unless Enum.empty?(@keywords) do %>
-  <table class="table">
-    <thead>
-      <tr>
-        <th>Name</th>
-        <th>Status</th>
-        <th># Links Count</th>
-        <th># Top-Adwords</th>
-        <th># Non-Ad</th>
-      </tr>
-    </thead>
-    <tbody>
-  <%= for keyword <- @keywords do %>
-    <tr>
-      <td><%= link keyword.name, to: Routes.keyword_path(@conn, :show, keyword.id) %></td>
-      <td><%= keyword.status %></td>
-      <%= if keyword.result do %>
-        <td class="count"><%= keyword.result.links_count %></td>
-        <td class="count"><%= keyword.result.top_ads |> Enum.count() %></td>
-        <td class="count"><%= keyword.result.non_ads |> Enum.count() %></td>
+        <div class="form-group">
+          <%= label :min_links_count, "Links Count(Min)" %>
+          <%= select f, :min_links_count, links_count_select_options() %>
+        </div>
+
+        <div class="form-group">
+          <%= label :max_links_count, "Links Count(Max)" %>
+          <%= select f, :max_links_count, links_count_select_options() %>
+        </div>
+
+        <%= submit "Filter" %>
+      <% end %>
+    </div>
+
+    <div class="col-9">
+      <%= unless Enum.empty?(@keywords) do %>
+        <h3> Keywords Listing </h3>
+        <table class="table">
+          <thead>
+            <tr>
+              <th>Name</th>
+              <th>Status</th>
+              <th>Links Count</th>
+              <th>Top-Ads</th>
+              <th>Non-Ads</th>
+            </tr>
+          </thead>
+          <tbody>
+        <%= for keyword <- @keywords do %>
+          <tr>
+            <td><%= link keyword.name, to: Routes.keyword_path(@conn, :show, keyword.id) %></td>
+            <td><%= keyword.status %></td>
+            <%= if keyword.result do %>
+              <td class="count"><%= keyword.result.links_count %></td>
+              <td class="count"><%= keyword.result.top_ads |> Enum.count() %></td>
```

Comment:
Let's use the function directly if it only single statement
```suggestion
              <td class="count"><%=  Enum.count(keyword.result.top_ads) %></td>
```

Reply:
Fixed